src/components/TextTruncated/index.tsx

```diff
@@ -0,0 +1,127 @@
+import { Link, Theme, Tooltip, Typography, TypographyProps } from '@mui/material';
+import { useState } from 'react';
+import { makeStyles } from '@mui/styles';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  arrow: {
+    color: `${theme.palette.TwClrBg} !important`,
+    '&:before': {
+      border: `1px solid ${theme.palette.TwClrBrdrTertiary}`
+    },
+  },
+  tooltip: {
+    backgroundColor: `${theme.palette.TwClrBg} !important`,
+    color: `${theme.palette.TwClrTxt} !important`,
+    border: `1px solid ${theme.palette.TwClrBrdrTertiary} !important`,
+    borderRadius: '8px !important',
+    boxShadow: `0px 3px 3px ${theme.palette.TwClrBaseGray100}`,
+    padding: '8px',
+  },
+}));
+
+export interface Props {
+  stringList: string[];
+  maxChars: number;
+  moreSeparator?: string;
+  moreText?: string;
+  textStyle?: TypographyProps['sx'];
+  showAllStyle?: TypographyProps['sx'];
+}
+
+/**
+ * Represents the result of truncating the string list
+ * text: the text to display before the '...# more' link
+ * numberMore: the number to display in the '...# more' link
+ *   has two special values
+ *   0: the full text can be displayed; do not show '...# more'
+ *   -1: the list is of length one but the element cannot be fully
+ *       displayed. show '...more' without a '#'
+ */
+interface TextDisplay {
+  text: string;
+  numberMore: number;
+}
+
+/**
+ * Compute TextDisplay for a list of strings
+ * @param strings list of strings to process
+ * @param maxLength max displayable length
+ */
+function computeFromStringList(
+  strings: string[],
+  maxLength: number,
+): TextDisplay {
+  if (strings[0].length > maxLength) {
+    // case in which the first string is too long to fit by itself
+    return {
+      text: strings[0].substring(0, maxLength),
+      numberMore: strings.length === 1 ? -1 : strings.length - 1
+    };
+  }
+
+  // concatenate strings with ', ' and stop before exceeding max
+  let stringsRemaining = strings.length - 1;
+  let result = strings[0];
+  for (let i = 1; i < strings.length; i++) {
+    const tempResult = [result, strings[i]].join(', ');
+    if (tempResult.length > maxLength) {
+      break;
+    }
+    stringsRemaining--;
+    result = tempResult;
+  }
+
+  return {
+    text: result,
+    numberMore: stringsRemaining,
+  };
+}
+
+export default function TextTruncated(props: Props): JSX.Element {
```

Comment:
extract props into a type?

Reply:
Do you mean doing 
``` typescript
export default function TextTruncated({prop1, prop2, prop3}: Props): JSX.Element {
```
instead?

Reply:
I do [define and export them here](https://github.com/terraware/web-components/pull/107/files#diff-fc13afc62e6095de3b24d5ee26ace1408c6165ab043f7c7d73625b7ccd8954ceR22). I'm sorry for my confusion not trying to be dense 😅

Reply:
gripe, my bad.. not sure where my confusion came from.. we don't have a codestyle guideline yet, but.. relocate interface to just above the component function?

Reply:
can do!